### PR TITLE
No one rack should have all that power

### PR DESCRIPTION
**Pull request by brendancheng20**
set up the endpoint for getting power information for a given asset

### CODE REVIEW

File: rackcity/views/pdu_views.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+from rackcity.api.serializers import (
+    serialize_power_connections,
+)
+# from rackcity.utils.log_utils import (
+#     log_action,
+#     log_bulk_import,
+#     log_delete,
+#     Action,
+#     ElementType,
+# )
```
Comment (benauriemma) [inline rackcity/views/pdu_views.py]: Can you delete commented imports

File: rackcity/views/pdu_views.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+# from rest_framework.parsers import JSONParser
+from http import HTTPStatus
+from rackcity.views.rackcity_utils import (
+    validate_asset_location,
+    validate_location_modification,
+    no_infile_location_conflicts,
+    records_are_identical,
+    get_sort_arguments,
+    get_filter_arguments,
+)
```
Comment (benauriemma) [inline rackcity/views/pdu_views.py]: And any unused ones as well

File: rackcity/views/pdu_views.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+import re
+import requests
+
+pdu_url = 'http://hyposoft-mgt.colab.duke.edu:8000/pdu.php?pdu=hpdu-rtp1-' # Need to specify rack + side in request, e.g. for A1 left, use A01L
```
Comment (benauriemma) [inline rackcity/views/pdu_views.py]: Ours is 8005 - do we need to use 8000 to test?

Comment (brendancheng20) [reply rackcity/views/pdu_views.py]: Using 8000 for testing purposes bc 8005 doesn't work when I run locally. I suppose we should change it now and see what happens on dev

File: rackcity/views/pdu_views.py (hunk 2 above)
Comment (brendancheng20) [reply rackcity/views/pdu_views.py]: Will do, totally forgot about this stuff

File: rackcity/views/pdu_views.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+            {"failure_message": failure_message},
+            status=HTTPStatus.BAD_REQUEST
+        )
+    port_info = serialize_power_connections(asset)
+
+    # get string parameter representing rack number (i.e. A01<L/R>)
+    rack_str = str(asset.rack.row_letter)
+    if (asset.rack.rack_num / 10 < 1):
+        rack_str = rack_str + "0"
+    rack_str = rack_str + str(asset.rack.rack_num)
```
Comment (benauriemma) [inline rackcity/views/pdu_views.py]: I think you can do like a %s thing to format the number into the string but this is fine

File: rackcity/views/pdu_views.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+    power_status = dict()
+    for port in port_info:
```
Comment (benauriemma) [inline rackcity/views/pdu_views.py]: Could you rename `port_info` to `power_connections` and this to `power_connection in power_connections`? I was a little confused by the name

File: rackcity/views/pdu_views.py [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+            "power_connections": port_info,
+            "power_status": power_status
+        },
+        status=HTTPStatus.OK
+    )
+
+
+def regex_power_status(html, port):
+    status_pattern = re.search('<td>'+str(port)+'<td><span.+(ON|OFF)', html)
+    return status_pattern.groups(1)
```
Comment (benauriemma) [inline rackcity/views/pdu_views.py]: The meat of this is an absolute beauty

**Pull request merged**